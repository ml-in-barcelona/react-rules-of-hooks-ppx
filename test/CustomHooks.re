let useClickOutside = fn => {
  if (true) {
    let elementRef = React.useRef(Js.Nullable.null);
  } else {
    ()
  }

  elementRef;
};
