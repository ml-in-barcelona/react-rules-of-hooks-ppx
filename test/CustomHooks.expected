[1mFile "CustomHooks.ml", line 3, characters 29-58[0m:
3 |     then (((let elementRef = React.useRef Js.Nullable.null in ()))
                                 [1;31m^^^^^^^^^^^^^^^^^^^^^^^^^^^^^[0m
[1;31mError[0m: Hooks can't be inside conditionals, neither loops.
